iOS_Project/Module/Home/View/HomeViewController.swift

```diff
@@ -43,12 +88,72 @@ class HomeViewController: UIViewController , UITableViewDelegate,UITableViewData
         cell?.setUpCell()
         cell?.catergoryName.text = categoryNames[indexPath.row]
         cell?.catergoryImage.image = UIImage(named: categoryNames[indexPath.row])
+        cell?.catergoryImage.layer.cornerRadius = 15
+        cell?.layer.cornerRadius = 15
 
+        changeCellBackground(cell: cell!)
         return cell ?? UICollectionViewCell()
     }
     
-    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
-        return 180
+    func changeCellBackground(cell : CategoryCollectionViewCell){
+        
+        if selectedCategory == cell.catergoryName.text{
+            cell.catergoryImage.backgroundColor = UIColor.orange
+        }else{
+            cell.catergoryImage.backgroundColor = UIColor.white
+        }
+    }
+    
+    func fetchingData(){
+        switch selectedCategory{
+            case "Popular":
+                homeViewModel?.fetchHomeData(tag: "middle_eastern")
```

Comment:
please add a function to ignore the redundancy

Reply:
comments resolved